### PR TITLE
Do not unintentionally send incorrect messages from history

### PR DESCRIPTION
**Pull request by xPaw**
Fixes the bug @astorije found.

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -403,14 +403,29 @@ Client.prototype.inputLine = function(data) {
...
-	var chan = target.chan;
-	var index = chan.messages.findIndex((val) => val.id === data.lastId);
-	var messages = chan.messages.slice(Math.max(0, index - 100), index);
+
+	const chan = target.chan;
+	const index = chan.messages.findIndex((val) => val.id === data.lastId);
+
+	// If we don't find the requested id, or it's first message we have
+	// Send an empty message array
+	if (index < 1) {
```
Comment (YaManicKill) [inline src/client.js]: Oh duh, I didn't even think about that possibility. Good catch.

File: client/js/socket-events/more.js [hunk 2]
```diff
@@ -16,6 +15,12 @@ socket.on("more", function(data) {
 	const scrollable = chan.closest(".chat");
 	const heightOld = chan.height();
 
+	// If there are no more messages to show, just hide the button and do nothing else
+	if (!data.messages.length) {
```
Comment (YaManicKill) [inline client/js/socket-events/more.js]: Do we need to take into consideration when we get more messages and previous ones will be pushed off the client?

File: src/client.js [hunk 3]
```diff
@@ -403,14 +403,29 @@ Client.prototype.inputLine = function(data) {
...
+		client.emit("more", {
+			chan: chan.id,
+			messages: []
+		});
+
+		return;
+	}
+
+	const messages = chan.messages.slice(Math.max(0, index - 100), index);
```
Comment (YaManicKill) [inline src/client.js]: Could we simplify this code by just doing

`const messages = index < 1 ? [] : chan.messages.slice(Math.max(0, index - 100), index);`

?

File: client/js/socket-events/more.js (hunk 2 above)
Comment (xPaw) [inline client/js/socket-events/more.js]: What do you mean?

File: src/client.js (hunk 3 above)
Comment (xPaw) [inline src/client.js]: Sure.

File: client/js/socket-events/more.js (hunk 2 above)
Comment (YaManicKill) [inline client/js/socket-events/more.js]: Well, if we have 100 messages in a channel, and we receive a new message, the top messages gets removed from the client, doesn't it?

Comment (xPaw) [inline client/js/socket-events/more.js]: Messages in active channel do not get trimmed.

Comment (YaManicKill) [inline client/js/socket-events/more.js]: Ah right, then no :-) cool

**Pull request merged**